pages/admin.js

```diff
@@ -0,0 +1,143 @@
+import Head from "next/head";
+import db from "../data/database";
+import { products as Products } from "../data/products";
+import { ListProductComponent } from "../components/product/list";
+import { useState, useEffect, useCallback } from "react";
+
+
+export default function Admin() {
+  console.log(Products)
+  let [product, setProduct] = useState({
+    name: "",
+    price: "",
+    description: "",
+    image: "",
+    flag: "",
+  });
+  const handleChange = (e) => {
+    setProduct({
+      ...product,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const submitNewProduct = (e) => {
+    e.preventDefault();
+    let priceNumber = parseFloat(product.price);
+    let admin = "Admin";
+    let productImage = `https://source.unsplash.com/300x300?${product.name}`;
+    if (
+      product.name !== "" &&
+      product.price !== "" &&
+      product.description !== ""
+    ) {
+      let newProduct = {
+        name: product.name,
+        description: product.description,
+        price: priceNumber,
+        image: productImage.toString(),
+        flag: admin,
+      };
+      db.table("products").add({
+        name: product.name,
+        description: product.description,
+        price: priceNumber,
+        image: productImage.toString(),
+        flag: admin,
```

Comment:
like 35-39 is the exact same as line 42-46. whenever you see duplication like this, you should refactor the code into a function.